openapi/tests/openapi_integration/test_geo_filter.py

```diff
@@ -0,0 +1,134 @@
+import random
+
+import pytest
+
+from .helpers.helpers import request_with_validation
+from .helpers.collection_setup import geo_collection_setup, drop_collection
+
+collection_name = "test_collection_filter"
+
+
+@pytest.fixture(autouse=True, scope="module")
+def setup():
+    geo_collection_setup(collection_name=collection_name)
+    yield
+    drop_collection(collection_name=collection_name)
+
+
+def test_geo_polygon():
```

Comment:
I'd love a few examples with malformed polygons to check they are detected and reported properly

Reply:
added.